prologin/workernode/operations.py

```diff
@@ -46,308 +50,319 @@ def untar(content, path, compression='gz'):
+        return Path('/box')
+
+    async def _run(
+        self,
+        *,
+        sockets_dir,
+        req_addr,
+        sub_addr,
+        champion_id,
+        player_id,
+        champion_tgz_b64,
+        map_content=None,
+        order_id=None,
+    ):
+        self.result['champion_id'] = champion_id
+
+        self.allowed_dirs.append(sockets_dir + ':rw')
+
+        champion_path = self.extract_champion(champion_tgz_b64)
+        env = {'CHAMPION_PATH': champion_path, 'HOME': '/tmp'}
+
+        # fmt: off
+        cmd = [
+            self.config['path']['stechec_client'],
+            "--name", str(player_id),
+            "--rules", self.config['path']['rules'],
+            "--champion", str(champion_path / 'champion.so'),
+            "--req_addr", req_addr,
+            "--sub_addr", sub_addr,
+            "--socket_timeout", "45000",
+            "--time", "1500",
+            "--verbose", "1",
+        ]
+        # fmt: on
+        if order_id is not None:
+            cmd += ["--client_id", str(order_id)]
+        if map_path := self.write_map(map_content):
+            cmd += ['--map', str(map_path)]
+
+        # Run the isolated client
+        await self.isolator.run(cmd, env=env, merge_outputs=True)
+
+
+async def compile_champion(config, champion_tgz_b64):
+    compile_champion = CompileChampion(config)
+    return await compile_champion(champion_tgz_b64=champion_tgz_b64)
+
+
+async def spawn_match(config, players, map_content):
     # Build the domain sockets
-    socket_dir = tempfile.TemporaryDirectory(prefix=f'workernode-{match_id}-')
+    socket_dir = tempfile.TemporaryDirectory(prefix='workernode-match-')
     os.chmod(socket_dir.name, 0o777)
     f_reqrep = socket_dir.name + '/' + 'reqrep'
     f_pubsub = socket_dir.name + '/' + 'pubsub'
     s_reqrep = 'ipc://' + f_reqrep
     s_pubsub = 'ipc://' + f_pubsub
 
     # Server task
-    task_server = asyncio.Task(
+    spawn_server = SpawnServer(config)
+    task_server = asyncio.create_task(
         spawn_server(
-            config,
-            s_reqrep,
-            s_pubsub,
-            len(players),
-            socket_dir.name,
-            map_contents,
+            sockets_dir=socket_dir.name,
+            rep_addr=s_reqrep,
+            pub_addr=s_pubsub,
+            nb_players=len(players),
+            map_content=map_content,
         )
     )
     await asyncio.sleep(0.1)  # Let the server start
 
     # Retry every seconds for 5 seconds
-    for i in range(5):
-        if all(os.access(f, os.R_OK | os.W_OK) for f in (f_reqrep, f_pubsub)):
-            break
-        await asyncio.sleep(1)
-    else:
-        raise RuntimeError("Server socket was never created")
+    @retry(reraise=True, stop=stop_after_attempt(7), wait=wait_fixed(2))
+    async def wait_for_server_sockets():
+        if not all(
+            os.access(f, os.R_OK | os.W_OK) for f in (f_reqrep, f_pubsub)
+        ):
+            raise RuntimeError("Server socket was never created")
```

Comment:
With this "never" it looks like you're wording the error knowing `@retry` will raise this eventually after a few failed attempts. But it makes reading the code weird.